src/storage/file_handle.cpp

```diff
@@ -44,20 +45,24 @@ void FileHandle::constructExistingFileHandle(const std::string& path, VirtualFil
     while (pageCapacity < numPages) {
         pageCapacity += StorageConstants::PAGE_GROUP_SIZE;
     }
-    loadFreeChunkMap();
 }
 
 void FileHandle::constructNewFileHandle(const std::string& path) {
     fileInfo = std::make_unique<FileInfo>(path, nullptr);
     numPages = 0;
     pageCapacity = 0;
-    loadFreeChunkMap();
 }
 
-void FileHandle::loadFreeChunkMap() {
-    freeChunkMap = std::make_unique<FreeChunkMap>();
+void FileHandle::loadFreeChunkMap(Deserializer& deserializer) {
+    std::unique_ptr<FreeChunkMap> deserializedFCM = FreeChunkMap::deserialize(deserializer);
+    if (freeChunkMap != nullptr) {
+        /* Extra precaution - if the map is nullptr for any reason, allow it to remain empty */
```

Comment:
Why are we allowing it to have an nullptr freeChunkMap?  Constructor of Filehandle should make sure that it is not nullptr; otherwise, report error or KU_ASSERT there

Reply:
Also, logic here should be assigning returned deserializedFCM to freeChunkMap of FileHandle. Please reconsider how should we properly initialize freeChunkMap here (i.e. change static FreeChunkMap::deserialize to be non-static?)

Reply:
Thanks for pointing this out. The check should have been on deserializedFCM, not freeChunkMap. Updated the code in the next iteration to make `FreeChunkMap::deserialize()` non-static so reassignment check is no longer relevant.